internal/tasks/fleet_rollout.go

```
@@ -6,23 +6,46 @@ import (
 	"github.com/flightctl/flightctl/pkg/log"
 	"github.com/flightctl/flightctl/pkg/reqid"
 	"github.com/go-chi/chi/v5/middleware"
-	"github.com/google/uuid"
 	"github.com/sirupsen/logrus"
 )
 
-/* Wait to be notified via channel about fleet template updates, exit upon ctx.Done() */
+// Wait to be notified via channel about fleet template updates, exit upon ctx.Done()
 func FleetRollouts(taskManager TaskManager) {
-	reqid.OverridePrefix("fleet-rollout")
-
 	for {
 		select {
-		case fleetRef := <-taskManager.channels[FleetTemplateRollout]:
-			err := rolloutFleet(taskManager.log, taskManager.store.Fleet(), taskManager.store.Device(), fleetRef)
-			if err != nil {
-				taskManager.log.Errorf("failed rolling out fleet %s/%s: %v", fleetRef.OrgID, fleetRef.Name, err)
+		case resourceRef := <-taskManager.channels[ChannelFleetTemplateRollout]:
+			requestID := reqid.NextRequestID()
+			ctx := context.WithValue(context.Background(), middleware.RequestIDKey, requestID)
+			log := log.WithReqIDFromCtx(ctx, taskManager.log)
+			logic := FleetRolloutsLogic{
+				ctx:         ctx,
+				log:         log,
+				fleetStore:  taskManager.store.Fleet(),
+				devStore:    taskManager.store.Device(),
+				resourceRef: resourceRef,
+			}
+
+			if resourceRef.Op != FleetRolloutOpUpdate {
+				taskManager.log.Errorf("received unknown op %s", resourceRef.Op)
+				break
+			}
+			if resourceRef.Kind == model.FleetKind {
+				err := logic.rolloutFleet()
+				if err != nil {
+					taskManager.log.Errorf("failed rolling out fleet %s/%s: %v", resourceRef.OrgID, resourceRef.Name, err)
+				}
+			} else if resourceRef.Kind == model.DeviceKind {
+				err := logic.rolloutDevice()
+				if err != nil {
+					taskManager.log.Errorf("failed rolling out device %s/%s: %v", resourceRef.OrgID, resourceRef.Name, err)
+				}
+			} else {
+				taskManager.log.Errorf("FleetRollouts called with incorrect resource kind %s", resourceRef.Kind)
 			}
 		case <-taskManager.ctx.Done():
```

I think we want to out this case at the top so we fail fast. WDYT?